lib/diaspora/message_renderer.rb

```diff
@@ -50,7 +50,10 @@ def escape
       end
 
       def strip_markdown
-        renderer = Redcarpet::Markdown.new Redcarpet::Render::StripDown, options[:markdown_options]
+        stripdown_options = options[:markdown_options]
+        # Footnotes are not supported in text-only outputs (mail, crossposts etc)
+        stripdown_options[:footnotes] = false
```

Comment:
This actually changes the value in `options` since it's only a reference to the original options. I don't know if it's a problem right now (if the same message renderer is uses twice), but I think it's at least an unexpected side-effect. I'd just do `stripdown_options = options[:markdown_options].except(:footnotes)` which gives you a new hash without that key, but without changing the original.

Reply:
Thanks for the code-suggestion and explanation ;-)